src/protocols/memory_attributes.rs

```diff
@@ -0,0 +1,40 @@
+//! Memory Attributes Protocol
+//!
+//! Provides an interface to abstract setting or getting of memory attributes in the UEFI environment.
+
+pub const PROTOCOL_GUID: crate::base::Guid = crate::base::Guid::from_fields(
+    0xf4560cf6,
+    0x40ec,
+    0x4b4a,
+    0xa1,
+    0x92,
+    &[0xbf, 0x1d, 0x57, 0xd0, 0xb1, 0x89],
+);
+
+pub type GetMemoryAttributes = eficall! {fn(
+    *const Protocol,
```

Comment:
We use `*mut` everywhere for now. Can you adjust the other ones as well? (see also the base docs for a rationale, if interested)

Reply:
Updated all instances.